ir/src/main/scala/org/scalajs/core/ir/Utils.scala

```diff
@@ -50,22 +52,46 @@ object Utils {
     /* Note that Java and JavaScript happen to use the same encoding for
      * Unicode, namely UTF-16, which means that 1 char from Java always equals
      * 1 char in JavaScript. */
-    val builder = new StringBuilder
-    str foreach {
-      case '\\'     => builder.append("\\\\")
-      case '"'      => builder.append("\\\"")
-      case '\u0007' => builder.append("\\a")
-      case '\u0008' => builder.append("\\b")
-      case '\u0009' => builder.append("\\t")
-      case '\u000A' => builder.append("\\n")
-      case '\u000B' => builder.append("\\v")
-      case '\u000C' => builder.append("\\f")
-      case '\u000D' => builder.append("\\r")
-      case c =>
-        if (c >= 32 && c <= 126) builder.append(c.toChar) // ASCII printable characters
-        else builder.append(f"\\u$c%04x")
+    val end = str.length()
+    var i = 0
+    while (i < end) {
+      val c = str.charAt(i)
+      if (c < 32 || c > 126)
+        i = end + 1
+      i += 1
+    }
+    if (i == end) {
+      str
+    } else {
+      def createEscapeJSString(str: String): String = {
+        val end = str.length()
+        val builder = new StringBuilder
+        builder.ensureCapacity(end)
+        var i = 0
+        while (i < end) {
+          val c = str.charAt(i)
+          if (c >= 32 && c <= 126) {
+            builder.append(c) // ASCII printable characters
+          } else {
+            builder.append((c: @switch) match {
+              case '\\' => "\\\\"
+              case '"' => "\\\""
+              case '\u0007' => "\\a"
+              case '\u0008' => "\\b"
+              case '\u0009' => "\\t"
+              case '\u000A' => "\\n"
+              case '\u000B' => "\\v"
+              case '\u000C' => "\\f"
+              case '\u000D' => "\\r"
+              case c => f"\\u$c%04x"
```

Comment:
Maybe extract that last `f` format in a tiny `def`, since it's a very rare case (much more so than `\`, `"` and `\n`, for example), and it consumes quite a lot of bytecode despite its source size.